worker/dbaccessor/worker.go

```diff
@@ -516,43 +533,64 @@ func (w *dbWorker) processAPIServerChange(apiDetails apiserver.Details) error {
 	ctx, cancel := w.scopedContext()
 	defer cancel()
 
-	// If this is an existing node, check if we are bound to the loopback IP
-	// and entering HA, which requires a new local-cloud bind address.
 	if extant {
 		asBootstrapped, err := mgr.IsBootstrappedNode(ctx)
 		if err != nil {
 			return errors.Trace(err)
 		}
 
-		// If this is a node that is already clustered, there is nothing to do.
-		if !asBootstrapped {
-			w.cfg.Logger.Debugf("node already clustered; no work to do")
-			return nil
+		serverCount := len(apiDetails.Servers)
+
+		// If we are as-bootstrapped, check if we are entering HA and need to
+		// change our binding from the loopback IP to a local-cloud address.
+		if asBootstrapped {
+			if serverCount == 1 {
+				// This bootstrapped node is still the only one around.
+				// We don't need to do anything.
+				return nil
+			}
+
+			addr, err := w.bindAddrFromServerDetails(apiDetails)
+			if err != nil {
+				if errors.Is(err, errors.NotFound) {
+					w.cfg.Logger.Infof(err.Error())
+					return nil
+				}
+				return errors.Trace(err)
+			}
+
+			if err := w.rebindAddress(ctx, addr); err != nil {
+				return errors.Trace(err)
+			}
+
+			log.Infof("successfully reconfigured Dqlite; restarting worker")
+			return dependency.ErrBounce
 		}
 
-		// If we are the as-bootstrapped node, and there is only one server,
-		// there is no need to change our bind address.
-		// Just keep the loopback binding.
-		if len(apiDetails.Servers) == 1 {
+		// If we are an existing, previously clustered node, but the node is
+		// not running, check if we're the only API server.
+		// If we are, then we have gone from HA-n to HA-1 and have a broken
+		// cluster. Attempt to reconfigure the cluster with just ourselves
+		// as a member.
```

Comment:
We could actually verify this by performing an open on the controller db.

Reply:
Not by the time we're here. The app is closed.